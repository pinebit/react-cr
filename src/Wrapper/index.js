import React from 'react';

const Wrapper = ({ children, wrapper, reverse, ...wrapperProps }) => {
  if (!children) {
    return null;
  }

  const useWrapper = React.Children.count(children) > 1 || wrapper;
  if (useWrapper) {
    const type = wrapper || 'div';
    const _children = reverse ? [...children].reverse() : children;
    return React.createElement(type, wrapperProps, _children);
  }

  return children;
};

Wrapper.propTypes = {
  wrapper: React.PropTypes.string,
  reverse: React.PropTypes.bool
};

export default Wrapper;
