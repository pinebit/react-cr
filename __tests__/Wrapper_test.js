import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../src/Wrapper';

it('Wrapper should reverse children when reverse is enabled', () => {
  const tree = renderer.create(
    <Wrapper reverse>
      <h1>Test1</h1>
      <h1>Test2</h1>
      <h1>Test3</h1>
    </Wrapper>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Wrapper should not reverse children when reverse is undefined', () => {
  const tree = renderer.create(
    <Wrapper>
      <h1>Test1</h1>
      <h1>Test2</h1>
      <h1>Test3</h1>
    </Wrapper>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Wrapper does not render when children is null', () => {
  const tree = renderer.create(
    <Wrapper>
      {null}
    </Wrapper>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
