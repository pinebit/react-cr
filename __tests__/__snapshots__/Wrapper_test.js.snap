// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper does not render when children is null 1`] = `null`;

exports[`Wrapper should not reverse children when reverse is undefined 1`] = `
<div>
  <h1>
    Test1
  </h1>
  <h1>
    Test2
  </h1>
  <h1>
    Test3
  </h1>
</div>
`;

exports[`Wrapper should reverse children when reverse is enabled 1`] = `
<div>
  <h1>
    Test3
  </h1>
  <h1>
    Test2
  </h1>
  <h1>
    Test1
  </h1>
</div>
`;
